pkg/operator/controller/dns/controller_test.go

```diff
@@ -0,0 +1,66 @@
+package dns
+
+import (
+	"github.com/google/go-cmp/cmp"
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+	"strings"
+	"testing"
+)
+
+func TestPublishRecordToZones(t *testing.T) {
+	var tests = []struct {
+		name   string
+		zones  []configv1.DNSZone
+		expect []string
+	}{
+		{
+			name: "testing for a successful update of 1 record",
+			zones: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			expect: []string{string(operatorv1.ConditionFalse)},
+		},
+		{
+			name:   "when no zones available",
+			zones:  []configv1.DNSZone{},
+			expect: []string{""},
+		},
+		{
+			name: "when one zone available and one not available",
+			zones: []configv1.DNSZone{
+				{ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+				{ID: ""}},
+			expect: []string{string(operatorv1.ConditionFalse), string(operatorv1.ConditionTrue)},
+		},
+	}
+
+	var conditions []string
```

Comment:
Declare `conditions` inside the `for` loop and delete `conditions = nil` inside the loop.

Reply:
done